### PR TITLE
[keyvault] azsecret renames

### PR DESCRIPTION
**Pull request by gracewilcox**
Renames for the azsecrets module

### CODE REVIEW

File: sdk/security/keyvault/azsecrets/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,16 @@
...
+## 1.0.0 (Unreleased)
+
+### Breaking Changes
+* Renamed `Client.ListSecrets` to `Client.ListSecretProperties`
+* Renamed `Client.ListSecretVersions` to `Client.ListSecretPropertiesVersions`
+* Renamed `SecretBundle` to `Secret`
+* Renamed `DeletedSecretBundle` to `DeletedSecret`
+* Renamed `SecretItem` to `SecretProperties`
+* Renamed `DeletedSecretItem` to `DeletedSecretProperties`
+* Renamed `Kid` to `KeyID`
```
Comment (chlowell) [inline sdk/security/keyvault/azsecrets/CHANGELOG.md:12]: I think we want `KID` instead. Key Vault represents keys as JWKs, in which "kid" is the correct name.

File: sdk/security/keyvault/azsecrets/autorest.md [hunk 2]
```diff
@@ -17,6 +17,17 @@ security-scopes: "https://vault.azure.net/.default"
 use: "@autorest/go@4.0.0-preview.46"
 version: "^3.0.0"
 
+declare-directive:
+  rename-property: >-
```
Comment (chlowell) [inline sdk/security/keyvault/azsecrets/autorest.md:21]: Please add a comment on this one, I can't tell what it does 😆

Comment (gracewilcox) [reply sdk/security/keyvault/azsecrets/autorest.md]: removed in [da9a800](https://github.com/Azure/azure-sdk-for-go/pull/20861/commits/da9a80082c721308944899e04818ff7ad034a6fd)

File: sdk/security/keyvault/azsecrets/autorest.md [hunk 3]
```diff
@@ -16,7 +16,7 @@ security: "AADToken"
 security-scopes: "https://vault.azure.net/.default"
 use: "@autorest/go@4.0.0-preview.46"
 version: "^3.0.0"
-
+
```
Comment (chlowell) [inline sdk/security/keyvault/azsecrets/autorest.md:19]:
```suggestion

```

**Review by chlowell — APPROVED** [review]

**Pull request merged**